### PR TITLE
Avoid redundant BatchNorm in create_head

### PR DESCRIPTION
**Pull request by renato145**
An attempt to solve the issue #3012 

I added a `first_bn` param in `create_head`.
And in `create_cnn_model`, it checks if the last layer is a `nn.BatchNorm2d` to set the `first_bn` param.

I though that may not be enough since some operations can be done after the batchnorm that are not necessarily a `nn.Module`, so I added a 2nd check to see if the output of the last batchnorm is the same as the output of the model.

### CODE REVIEW

**Comment by jph00** [comment]
This is super neat, but I think it's a bit to complex for this foundational function.

I like the changes to `create_head`, so let's keep those. But could you remove the changes in `create_cnn_model` so this becomes just a flag people pass in manually?

**Comment by renato145** [comment]
Ok :)

**Comment by renato145** [comment]
There were some warning I didn't notice before, I just cleaned them.

**Comment by jph00** [comment]
Many thanks!
